### ISSUE TITLE
Do not load all countries at first

### ISSUE THREAD

**Issue by m-avagyan** [goveo/react-international-phone#161]
**Countries list do not loaded lazyly.**
The countries all list loaded at first at dropdown.

**Intersection observer**
Using intersection observer we can get if latest element is visible and loaded next countries data.

**Describe alternatives you've considered**

**Performance issues**
This happen a excessive DOM size issue

**Comment by Sayvai** [comment]
I agree. Maybe a **virtualised** solution to load the list of countries works best(?)

There is noticeably perceptible ~2/3 second delay of rendering the list of countries for each subsequent click to render the countries, which is typically noticeable **after** the first render of the countries, only when i have Chrome Dev Tools open within the page.

#### Rendering the list of countries from dropdown select with Chrome Dev Tools open
![react-international-phone-countries-slow-render-with-dev-tools](https://github.com/goveo/react-international-phone/assets/7581546/9ac92afd-a2ca-4e4a-a59a-e2c37ef20e20)

However, when Chrome Dev Tools is not open, the countries list load performance is much better, although i can somewhat still notice a slight lag, which is just about acceptable(?), but definitely not optimal, and could be improved for a much more snappy user experience 🙁 

#### Rendering the list of countries from dropdown select with Chrome Dev Tools closed
![react-international-phone-countries-slow-render-no-dev-tools](https://github.com/goveo/react-international-phone/assets/7581546/b6b01694-8b05-471f-abdf-dcb029e5f95d)

I suppose it could also be machine resource dependent too, whether system resource usage (e.g. CPU) is peaking at the time of interaction.
